llvm/lib/Transforms/IPO/GlobalOpt.cpp

```diff
@@ -1901,6 +1902,16 @@ static void makeAllConstantUsesInstructions(Constant *C) {
   }
 }
 
+/// Reset the global varible GV TicketNode metadata.
+static void calculateGlobalDigest(GlobalVariable *GV) {
+  if (!GV->getMetadata(LLVMContext::MD_repo_ticket))
```

Comment:
I think that this is testing whether we're compiling to a repo. Is this the best way? Rather than searching the metadata, could we just check the triple? Could we do it once rather than every time a global is modified?

Reply:
> 
> 
> I think that this is testing whether we're compiling to a repo. Is this the best way? Rather than searching the metadata, could we just check the triple?
> 

Good point. I have updated the patch using the triple.

>
> Could we do it once rather than every time a global is modified?
>

We want to re-calculate digest only the global variable initial value is modified. I checked LLVM build by printing the message when the initial value is changed. I found it is not too much that the global initial value is changed during the globalopt. Without any unnecessary recalculations, I can't manage to do once only.

Reply:
> We want to re-calculate digest only the global variable initial value is modified.

Yes. I was asking about the code which decides if we're compiling to the repo. ATM we work this out every time a GV has been modified. Could we do it just once?

Reply:
Sorry to misunderstand you. Yes, it has been fixed.